sbtgen/src/main/scala/izumi/sbtgen/Renderer.scala

```diff
@@ -86,6 +86,26 @@ class Renderer(
   protected val index: Map[ArtifactId, Artifact] = makeIndex(aggregates)
   protected val settingsCache = new mutable.LinkedHashMap[String, Int]()
 
+  protected val configuredGroups: Set[String] = {
+    def flattenGroups(groups: Set[Group]): Set[Group] = {
+      groups.foldLeft(Set.empty: Set[Group])((agg, g) => agg ++ unpackGroup(g))
+    }
+
+    def unpackGroup(group: Group): Set[Group] = {
+      if (group.deps.isEmpty) Set(Group(group.name))
```

Comment:
```suggestion
      Set(group) ++ group.deps.flatMap(unpackGroup)
```
it's ok for the first group to keep its deps, then it can be equal to groups in user data without extracting .name